prometheus-connector/src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusSinkBuilder.java

```diff
@@ -0,0 +1,203 @@
+                retryConfiguration.getInitialRetryDelayMS(),
+                retryConfiguration.getMaxRetryDelayMS(),
+                retryConfiguration.getMaxRetryCount(),
+                socketTimeoutMs);
+
+        return new PrometheusSink(
+                new PrometheusTimeSeriesConverter(),
+                maxInFlightRequest,
+                actualMaxBufferedRequests,
+                actualMaxBatchSizeInSamples,
+                actualMaxRecordSizeInSamples,
+                actualMaxTimeInBufferMS,
+                prometheusRemoteWriteUrl,
+                new PrometheusAsyncHttpClientBuilder(retryConfiguration)
+                        .setSocketTimeout(actualSocketTimeoutMs),
+                requestSigner);
+    }
+
+    private static void checkValidRemoteWriteUrl(String url) {
+        try {
+            new URL(url);
+        } catch (MalformedURLException mue) {
+            throw new IllegalArgumentException("Invalid Remote-Write URL: " + url, mue);
+        }
+    }
+
+    public PrometheusSinkBuilder setPrometheusRemoteWriteUrl(String prometheusRemoteWriteUrl) {
+        this.prometheusRemoteWriteUrl = prometheusRemoteWriteUrl;
+        return this;
+    }
+
+    public PrometheusSinkBuilder setRequestSigner(PrometheusRequestSigner requestSigner) {
+        this.requestSigner = requestSigner;
+        return this;
+    }
+
+    public PrometheusSinkBuilder setMaxBatchSizeInSamples(int maxBatchSizeInSamples) {
+        this.maxBatchSizeInSamples = maxBatchSizeInSamples;
+        return this;
+    }
+
+    public PrometheusSinkBuilder setMaxRecordSizeInSamples(int maxRecordSizeInSamples) {
+        this.maxRecordSizeInSamples = maxRecordSizeInSamples;
+        return this;
+    }
+
+    public PrometheusSinkBuilder setRetryConfiguration(RetryConfiguration retryConfiguration) {
+        this.retryConfiguration = retryConfiguration;
+        return this;
+    }
+
+    public PrometheusSinkBuilder setSocketTimeoutMs(int socketTimeoutMs) {
+        this.socketTimeoutMs = socketTimeoutMs;
+        return this;
+    }
+
+    private Integer getMaxBatchSizeInSamples() {
+        return maxBatchSizeInSamples;
+    }
+
+    private Integer getMaxRecordSizeInSamples() {
+        return maxRecordSizeInSamples;
+    }
+
+    public RetryConfiguration getRetryConfiguration() {
+        return retryConfiguration;
+    }
+
+    public Integer getSocketTimeoutMs() {
+        return socketTimeoutMs;
+    }
+
+    /// Disable setting maxBatchSize, maxBatchSizeInBytes, and maxRecordSizeInBytes directly
+
+    @Override
+    public PrometheusSinkBuilder setMaxBatchSize(int maxBatchSize) {
+        throw new UnsupportedOperationException("maxBatchSize is not supported by this sink");
+    }
+
+    @Override
+    public PrometheusSinkBuilder setMaxBatchSizeInBytes(long maxBatchSizeInBytes) {
+        throw new UnsupportedOperationException(
+                "maxBatchSizeInBytes is not supported by this sink");
```

Comment:
Should we add a comment on the reason why we disable it here?

Reply:
After offline discussion, maybe we can do something like 

```
Use setMaxBatchSizeInSamples() instead
```

Reply:
Added comments to all disabled parameters, explaining what to use instead